### PR TITLE
Axial tilt

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: journal/player_test.cpp [hunk 1]
```diff
@@ -93,14 +93,14 @@ TEST_F(PlayerTest, Benchmarks) {
   }
 }
 
-#if 0
+#if 1
```
Comment (pleroy) [inline journal/player_test.cpp]: Change back to 0.

File: journal/player_test.cpp [hunk 2]
```diff
@@ -93,14 +93,14 @@ TEST_F(PlayerTest, Benchmarks) {
...
 // This test is only run if the --gtest_filter flag names it explicitly.
 TEST_F(PlayerTest, Debug) {
   if (testing::FLAGS_gtest_filter == test_case_name_ + "." + test_name_) {
     // An example of how journalling may be used for debugging.  You must set
     // |path| and fill the |method_in| and |method_out_return| protocol buffers.
     std::string path =
-        R"(P:\Public Mockingbird\Principia\Journals\JOURNAL.20160525-191742)";
+        R"(C:\Users\robin\Projects\Kerbal Space Program\KSP_win 1.1.3\glog\Principia\JOURNAL.20160811-141757)";
```
Comment (pleroy) [inline journal/player_test.cpp]: Change back to a path rooted at `P:` otherwise it won't work on other machines.

File: journal/player_test.cpp [hunk 3]
```diff
@@ -114,14 +114,14 @@ TEST_F(PlayerTest, Debug) {
     LOG(ERROR) << "Last successful method out/return: \n"
                << player.last_method_out_return().DebugString();
 
-#if 0
+#if 1
```
Comment (pleroy) [inline journal/player_test.cpp]: 0

File: ksp_plugin/frames.hpp [hunk 4]
```diff
@@ -37,15 +37,19 @@ using AliceWorld = Frame<serialization::Frame::PluginTag,
 
 // The barycentric reference frame of the solar system.
 // The basis is the basis of |AliceWorld| at |UniversalTimeEpoch|.
-// TODO(egg): it *should* be the barycentric frame. For the moment we're using
-// the velocity of the sun at the time of construction as our reference.
-// The origin is the position of the sun at the instant |initial_time| passed at
-// construction.
-// TODO(phl): this has been flipped, change the value of the protobuf enum and
```
Comment (pleroy) [inline ksp_plugin/frames.hpp]: Please create an issue.  This needs to go in Cardano.

File: ksp_plugin/plugin.cpp [hunk 5]
```diff
@@ -235,6 +237,69 @@ void Plugin::UpdateCelestialHierarchy(Index const celestial_index,
...
+  Permutation<BodyWorld, BodyFixed> const body_mirror(
+      Permutation<BodyWorld, BodyFixed>::XZY);
+
+  auto const& body = dynamic_cast<RotatingBody<Barycentric> const&>(
+      *FindOrDie(celestials_, index)->body());
+
+  Bivector<double, BodyFixed> z({0, 0, 1});
+  Bivector<double, BodyFixed> x({1, 0, 0});
+
+  // TODO(egg): Euler angles.
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Reference #810.

File: ksp_plugin/plugin.cpp [hunk 6]
```diff
@@ -685,7 +750,7 @@ void Plugin::WriteToMessage(
       message->mutable_bubble());
 
   planetarium_rotation_.WriteToMessage(message->mutable_planetarium_rotation());
-  game_epoch_.WriteToMessage(message->mutable_game_epoch());
+  game_epoch_.WriteToMessage(message->mutable_game_epoch());
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Why the change?

File: ksp_plugin/plugin.cpp [hunk 7]
```diff
@@ -869,9 +936,29 @@ not_null<std::unique_ptr<Vessel>> const& Plugin::find_vessel_by_guid_or_die(
 // The map between the vector spaces of |Barycentric| and |AliceSun| at
 // |current_time_|.
 Rotation<Barycentric, AliceSun> Plugin::PlanetariumRotation() const {
-  return Rotation<Barycentric, AliceSun>(
-      planetarium_rotation_,
-      Bivector<double, Barycentric>({0, 0, -1}));
+  // The z axis of |PlanetariumFrame| is the pole of |main_body_|, and x axis is
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: ... and its x axis ...

File: ksp_plugin/plugin.cpp [hunk 8]
```diff
@@ -869,9 +936,29 @@ not_null<std::unique_ptr<Vessel>> const& Plugin::find_vessel_by_guid_or_die(
...
+  // plane and the plane of |main_body_|'s equator, or the y axis of
+  // |Barycentric| if they coincide).
+  // This can be expressed using Euler angles, see figures 1 and 2 of
+  // http://astropedia.astrogeology.usgs.gov/download/Docs/WGCCRE/WGCCRE2009reprint.pdf.
+  struct PlanetariumFrame;
+
+  Bivector<double, PlanetariumFrame> z({0, 0, 1});
+  Bivector<double, PlanetariumFrame> x({1, 0, 0});
+
+  // TODO(egg): this would be more clearly expressed using zxz euler angles (the
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Euler.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 9]
```diff
@@ -795,14 +834,41 @@ where is_in_physics_bubble(vessel)
     if (!PluginRunning()) {
       return;
     }
+
+    #if DRAW_BODY_AXES
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Do we need to keep this code?  Will it be useful in the future?

File: ksp_plugin_adapter/map_renderer.cs [hunk 10]
```diff
@@ -6,7 +6,7 @@
 namespace principia {
 namespace ksp_plugin_adapter {
 
-class MapRenderer : UnityEngine.MonoBehaviour {
+class RenderingActions : UnityEngine.MonoBehaviour {
```
Comment (pleroy) [inline ksp_plugin_adapter/map_renderer.cs]: Please rename the file to match the class.

File: serialization/geometry.proto [hunk 11]
```diff
@@ -138,6 +138,8 @@ message Frame {
     NAVIGATION = 4;
     WORLD = 5;
     WORLD_SUN = 6;
+    CELESTIAL_SPHERE = 8;
```
Comment (pleroy) [inline serialization/geometry.proto]: Alphabetize.

File: serialization/journal.proto [hunk 12]
```diff
@@ -168,6 +168,37 @@ message CelestialFromParent {
   optional Return return = 3;
 }
 
+message CelestialRotation {
```
Comment (pleroy) [inline serialization/journal.proto]: It would make me happier if we could find a systematic naming convention where all related methods start with the same word, and we have an `interface_foo.cpp` to hold their implementation.  `interface.cpp` is just too messy at this point.

Now we have to argue about the names.  `SetMainBody` is a bit special, so let's keep it as it is (and in `interface.cpp`).  All the other methods have `rotation` in their name.  A suggestion:
```
RotationOfCelestial
RotationOfCelestialSphere
RotationGetInitialInDegrees
RotationGetPeriod
```

File: journal/player_test.cpp (hunk 1 above)
Comment (eggrobin) [inline journal/player_test.cpp]: Resetting this file to master, there's no reason for things to change here.

File: journal/player_test.cpp (hunk 2 above)
Comment (eggrobin) [inline journal/player_test.cpp]: Same.

File: journal/player_test.cpp (hunk 3 above)
Comment (eggrobin) [inline journal/player_test.cpp]: Same.

File: ksp_plugin/frames.hpp (hunk 4 above)
Comment (eggrobin) [inline ksp_plugin/frames.hpp]: As far as I can tell the second TODO was precisely about `PRE_BOREL_BARYCENTRIC`, and was addressed before releasing Borel ([blame](https://github.com/mockingbirdnest/Principia/blame/5fcca7e/ksp_plugin/frames.hpp#L44), [commit](https://github.com/mockingbirdnest/Principia/commit/5c017d6f0b50394346119a1310e29a795d783839), note that it's in since Borel, [pull request containing the commit](https://github.com/mockingbirdnest/Principia/pull/495), note that it's the one that flips the universe).
The first one was incidentally addressed in #835, when switching to Jacobi coordinates for Keplerian elements.

File: ksp_plugin/plugin.cpp (hunk 5 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 6 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Not sure what happened here; the linter saw it, removed.

File: ksp_plugin/plugin.cpp (hunk 7 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 8 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 9 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: If it's useful we can just fetch it from the tree anyway. Removing.

File: ksp_plugin_adapter/map_renderer.cs (hunk 10 above)
Comment (eggrobin) [inline ksp_plugin_adapter/map_renderer.cs]: Done.

File: serialization/geometry.proto (hunk 11 above)
Comment (eggrobin) [inline serialization/geometry.proto]: Done.

File: serialization/journal.proto (hunk 12 above)
Comment (eggrobin) [inline serialization/journal.proto]: As discussed,
```
CelestialRotation
CelestialSphereRotation
CelestialInitialRotationInDegrees
CelestialRotationPeriod
```

File: ksp_plugin/plugin.cpp [hunk 13]
```diff
@@ -288,13 +288,13 @@ Rotation<CelestialSphere, World> Plugin::CelestialSphereRotation()
...
 
-Time Plugin::RotationPeriod(Index const celestial_index) const {
+Time Plugin::CelestialRotationPeriod(Index const celestial_index) const {
   auto const& body = dynamic_cast<RotatingBody<Barycentric> const&>(
       *FindOrDie(celestials_, celestial_index)->body());
   return 2 * π * Radian / body.angular_velocity().Norm();
 }
 
-Angle Plugin::InitialRotation(Index const celestial_index) const {
+Angle Plugin::CelestialInitialRotation(Index const celestial_index) const {
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Alphabetize wrt the previous method.

File: ksp_plugin/plugin.hpp [hunk 14]
```diff
@@ -121,8 +121,8 @@ class Plugin {
   virtual Rotation<BodyWorld, World> CelestialRotation(Index const index) const;
   virtual Rotation<CelestialSphere, World> CelestialSphereRotation() const;
 
-  virtual Time RotationPeriod(Index const celestial_index) const;
-  virtual Angle InitialRotation(Index const celestial_index) const;
+  virtual Time CelestialRotationPeriod(Index const celestial_index) const;
+  virtual Angle CelestialInitialRotation(Index const celestial_index) const;
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Alphabetize.

File: ksp_plugin_adapter/ksp_plugin_adapter.csproj [hunk 15]
```diff
@@ -74,7 +74,7 @@
     <Compile Include="loader.cs" />
     <Compile Include="logging.cs" />
     <Compile Include="map_node_pool.cs" />
-    <Compile Include="map_renderer.cs" />
+    <Compile Include="rendering_actions.cs" />
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.csproj]: Can this be alphabetized somehow?

Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.csproj]: Maybe, but I'd rather not fight the xml generator.

File: ksp_plugin/plugin.hpp (hunk 14 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 13 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin_test/plugin_test.cpp [hunk 16]
```diff
@@ -468,29 +484,35 @@ TEST_F(PluginTest, HierarchicalInitialization) {
...
   EXPECT_CALL(*mock_ephemeris_, WriteToMessage(_))
       .WillOnce(SetArgPointee<0>(valid_ephemeris_message_));
   plugin_->EndInitialization();
   EXPECT_CALL(*mock_ephemeris_, Prolong(_)).Times(AnyNumber());
   EXPECT_THAT(plugin_->CelestialFromParent(1).displacement().Norm(),
-              AlmostEquals(3.0 * Metre, 1));
+              AlmostEquals(3.0 * Metre, 3));
   EXPECT_THAT(plugin_->CelestialFromParent(2).displacement().Norm(),
-              Eq(1 * Metre));
+              AlmostEquals(1 * Metre, 2, 3));
```
Comment (pleroy) [inline ksp_plugin_test/plugin_test.cpp]: This seems wrong.  We should only use the 3-argument `AlmostEquals` in a loop, I think.

File: ksp_plugin_test/plugin_test.cpp [hunk 17]
```diff
@@ -806,15 +828,17 @@ TEST_F(PluginTest, UpdateCelestialHierarchy) {
                   2),
               AlmostEquals(
                   to_icrf(plugin_->CelestialFromParent(index).velocity()),
-                  20155840)))
+                  20159592)))
           << SolarSystemFactory::name(index);
     } else {
+      // TODO(egg): I'm not sure that's really fine, this seems to be 20 bits?
+      // and 24 on Triton.  What's going on?
```
Comment (pleroy) [inline ksp_plugin_test/plugin_test.cpp]: No idea what this comment means.

File: ksp_plugin/plugin.cpp [hunk 18]
```diff
@@ -750,7 +750,11 @@ void Plugin::WriteToMessage(
       message->mutable_bubble());
 
   planetarium_rotation_.WriteToMessage(message->mutable_planetarium_rotation());
-  game_epoch_.WriteToMessage(message->mutable_game_epoch());
+  if (!is_pre_cardano_) {
+    // A pre-Cardano save stays pre-Cardano; we cannot pull rotational
+    // properties out of thin air.
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: This is a problem.  It cannot really work that way, first because of the number of places that check `is_pre_cardano` but also because future compatibility code will be insanely messy.
